### PR TITLE
Generate runtime fallback graph for shared framework

### PR DESCRIPTION
**Pull request by pakrym**
@anurse

### CODE REVIEW

File: tools/RuntimeGraphGenerator/Program.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+using NuGet.Versioning;
+using Microsoft.DotNet.ProjectModel.Graph;
+
+namespace RuntimeGraphGenerator
+{
+    public class RuntimeGraphGeneratorCommand
```
Comment (anurse) [inline tools/RuntimeGraphGenerator/Program.cs]: Rename to match file: `Program`

Comment (pakrym) [inline tools/RuntimeGraphGenerator/Program.cs]: :+1:

File: tools/RuntimeGraphGenerator/Program.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                }
+                var framework = NuGetFramework.Parse(context.TargetFramework);
+                var projectContext = ProjectContext.Create(projectDirectory, framework);
+
+                // Configuration is used only for P2P dependencies so were don't care
+                var exporter = projectContext.CreateExporter("Debug");
+                var manager = new RuntimeGraphManager();
+                var graph = manager.Collect(exporter.GetDependencies(LibraryType.Package));
+                var expandedGraph = manager.Expand(graph, runtimes);
+                var graphManager = new RuntimeGraphManager();
```
Comment (anurse) [inline tools/RuntimeGraphGenerator/Program.cs]: not needed?

File: tools/RuntimeGraphGenerator/RuntimeGraphCollector.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+using System.Threading.Tasks;
+using Microsoft.DotNet.ProjectModel.Compilation;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.RuntimeModel;
+using System.IO;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.DotNet.ProjectModel
+{
+    public class RuntimeGraphManager
```
Comment (anurse) [inline tools/RuntimeGraphGenerator/RuntimeGraphCollector.cs]: Rename file or type to match :)

**Comment by anurse** [comment]
Minor things, but otherwise looks good! :shipit:

File: tools/RuntimeGraphGenerator/Program.cs (hunk 2 above)
Comment (pakrym) [inline tools/RuntimeGraphGenerator/Program.cs]: You can never get enough managers.

**Comment by pakrym** [comment]
https://github.com/dotnet/cli/issues/1688

**Pull request merged**